lib/streamlit/connections/sql_connection.py

```diff
@@ -221,6 +217,12 @@ def _query(
                 **kwargs,
             )
 
+        _query.__qualname__ = f"{_query.__qualname__}_{self._connection_name}_{ttl}"
```

Comment:
nit: we need to preprocess `ttl` here to handle `timedelta` and `float` types correctly, although looks like just formatting to string works correctly.

Reply:
hmm, from some manual testing in the Python REPL, it seems like both `float`s and `timedelta`s get serialized to `str` in a reasonable way when just inserting them into this format string